### DISCUSSION TITLE
How to integrate Keycloak?

### DISCUSSION THREAD

**Discussion by nemonemi** [keystonejs/keystone#8044] [Questions]
I would like to use the Keystonejs as a configuration CMS for my application, and for that, I would like to define users, roles, user groups etc. that will be affected by this configuration.

I would like to know if there is any plugin or an existing integration for the Keycloak?

Basically, Keystone would display the data from the Keycloak (those users, user groups, roles, etc.) for selection.

Thanks!

**Comment by flexdinesh** [comment]
Keystone supports [virtual fields](https://keystonejs.com/docs/fields/virtual). They're basically read-time fields where the data is fetched from an external REST/GraphQL API. We don't have an official intergration with Keycloak but you should be able to build one by mapping your schema with virtual fields to Keycloak API if they have one.
